ush/get_obs.py

```diff
@@ -0,0 +1,1004 @@
+        field_groups_in_obs = ['REFC', 'RETOP']
+    elif obtype == 'NDAS':
+        field_groups_in_obs = ['ADPSFCandADPUPA']
+    num_field_groups = len(field_groups_in_obs)
+
+    # For each field group in the observations, get the METplus file name
+    # template for the observation files.  Then combine these with the base
+    # directory to get the METplus template for the full path on disk to
+    # the processed obs files.  If obs files do not already exist at these
+    # locations, they will be retrieved from HPSS and placed at these locations.
+    fp_proc_templates = []
+    for fg in field_groups_in_obs:
+        key = 'OBS_' + obtype + '_' + fg + '_FN_TEMPLATE'
+        fn_proc_template = vx_config[key]
+        fp_proc_templates.append(os.path.join(obs_dir, fn_proc_template))
+    #
+    #-----------------------------------------------------------------------
+    #
+    # Set variables that are only needed for some obs types.
+    #
+    #-----------------------------------------------------------------------
+    #
+
+    # For cumulative obs, set the accumulation period to use when getting obs
+    # files.  This is simply a properly formatted version of the obs availability
+    # interval.
+    accum_obs_formatted = None
+    if obtype == 'CCPA':
+        accum_obs_formatted = f'{obs_avail_intvl_hrs:02d}'
+    elif obtype == 'NOHRSC':
+        accum_obs_formatted = f'{obs_avail_intvl_hrs:d}'
+
+    # For MRMS obs, set field-dependent parameters needed in forming grib2
+    # file names.
+    fields_in_filenames = []
+    levels_in_filenames = []
+    if obtype == 'MRMS':
+        for fg in field_groups_in_obs:
+            if fg == 'REFC':
+                fields_in_filenames.append('MergedReflectivityQCComposite')
+                levels_in_filenames.append('00.50')
+            elif fg == 'RETOP':
+                fields_in_filenames.append('EchoTop')
+                levels_in_filenames.append('18_00.50')
+            else:
+                msg = dedent(f"""
+                    Invalid field specified for obs type:
+                        obtype = {obtype}
+                        field = {field}
+                    """)
+                logging.error(msg)
+                raise Exception(msg)
+
+    # CCPA files for 1-hour accumulation have incorrect metadata in the files
+    # under the "00" directory from 20180718 to 20210504.  Set these starting
+    # and ending dates as datetime objects for later use.
+    yyyymmdd_bad_metadata_start_str = None
+    yyyymmdd_bad_metadata_end_str = None
+    yyyymmdd_bad_metadata_start = None
+    yyyymmdd_bad_metadata_end = None
+    if obtype == 'CCPA':
+        yyyymmdd_bad_metadata_start_str = '20180718'
+        yyyymmdd_bad_metadata_end_str = '20210504'
+        yyyymmdd_bad_metadata_start = dt.datetime.strptime(yyyymmdd_bad_metadata_start_str, '%Y%m%d')
+        yyyymmdd_bad_metadata_end = dt.datetime.strptime(yyyymmdd_bad_metadata_end_str, '%Y%m%d')
```

Comment:
This is only ever used for CCPA obs, this logic can be way simpler
```suggestion
    ccpa_bad_metadata_start = dt.datetime.strptime('20180718', '%Y%m%d')
    ccpa_bad_metadata_end = dt.datetime.strptime('20210504', '%Y%m%d')
```

Reply:
Changed.